test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/CompleteCaseControllerSpec.scala

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tariffclassificationfrontend.controllers
+
+import java.time.Clock
+
+import akka.stream.Materializer
+import org.mockito.ArgumentMatchers.{any, refEq}
+import org.mockito.Mockito._
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, Matchers, WordSpec}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import play.api.http.HeaderNames.LOCATION
+import play.api.http.{MimeTypes, Status}
+import play.api.i18n.{DefaultLangs, DefaultMessagesApi}
+import play.api.mvc.{AnyContentAsEmpty, AnyContentAsFormUrlEncoded, Result}
+import play.api.test.{FakeHeaders, FakeRequest}
+import play.api.{Configuration, Environment}
+import play.filters.csrf.CSRF.{Token, TokenProvider}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.UnitSpec
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.{CaseStatus, Operator}
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.tariffclassificationfrontend.utils.Cases
+
+import scala.concurrent.Future.{failed, successful}
+
+class CompleteCaseControllerSpec extends WordSpec with Matchers with UnitSpec
+  with GuiceOneAppPerSuite with MockitoSugar with BeforeAndAfterEach {
+
+  private val env = Environment.simple()
+
+  private val configuration = Configuration.load(env)
+  private val messageApi = new DefaultMessagesApi(env, configuration, new DefaultLangs(configuration))
+  private val appConfig = new AppConfig(configuration, env)
+  private val casesService = mock[CasesService]
+  private val operator = mock[Operator]
+
+  private val caseWithStatusOPEN = Cases.btiCaseExample.copy(status = CaseStatus.OPEN)
+  private val caseWithStatusCOMPLETED = Cases.btiCaseExample.copy(status = CaseStatus.COMPLETED)
+  private val caseWithoutDecision = Cases.btiCaseExample.copy(status = CaseStatus.OPEN, decision = None)
+
+  private implicit val mat: Materializer = app.materializer
+  private implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  private val controller = new CompleteCaseController(new SuccessfulAuthenticatedAction(operator), casesService, messageApi, appConfig)
+
+  override def beforeEach(): Unit = {
```

Comment:
I believe this is not needed if you do not add anything to the parent implementation.

Reply:
removed